src/onnxruntime_benchmark/src/inputs_preparation.cpp

```diff
@@ -0,0 +1,471 @@
+    logger::info << "\t\t" << image_size.width << "x" << image_size.height << logger::endl;
+    for (int b = 0; b < batch_size; ++b) {
+        int64_t image_info_size = tensor_size / batch_size;
+        for (int i = 0; i < image_info_size; ++i) {
+            int id = b * image_info_size + i;
+            if (0 == i) {
+                tensor_data[id] = static_cast<T>(image_size.width);
+            }
+            else if (1 == i) {
+                tensor_data[id] = static_cast<T>(image_size.height);
+            }
+            else {
+                tensor_data[id] = 1;
+            }
+        }
+    }
+    return tensor;
+}
+
+template <class T>
+Ort::Value create_tensor_from_binary(const inputs::InputDescr &input_descr, int batch_size, int start_index) {
+    auto tensor_descr = input_descr.tensor_descr;
+    const auto &files = input_descr.files;
+
+    auto allocator = Ort::AllocatorWithDefaultOptions();
+    auto tensor = Ort::Value::CreateTensor(allocator,
+                                           tensor_descr.data_shape.data(),
+                                           tensor_descr.data_shape.size(),
+                                           tensor_descr.type);
+
+    size_t tensor_size =
+        std::accumulate(tensor_descr.data_shape.begin(), tensor_descr.data_shape.end(), 1, std::multiplies<int64_t>());
+
+    auto tensor_data = tensor.GetTensorMutableData<char>();
+    for (int b = 0; b < batch_size; ++b) {
+        size_t input_id = (start_index + b) % files.size();
+        const auto &file_path = files[input_id];
+        logger::info << "\t\t" << file_path << logger::endl;
+
+        std::ifstream binary_file(file_path, std::ios_base::binary | std::ios_base::ate);
+        if (!binary_file) {
+            throw std::runtime_error("Can't open " + file_path);
+        }
+
+        auto file_size = static_cast<std::size_t>(binary_file.tellg());
+        auto input_size = tensor_size * sizeof(T) / batch_size;
+        if (file_size != input_size) {
+            throw std::invalid_argument("File " + file_path + " contains " + std::to_string(file_size) +
+                                        " bytes but the mdoel expects " + std::to_string(input_size));
+        }
+
+        binary_file.seekg(0, std::ios_base::beg);
+        if (!binary_file.good()) {
+            throw std::runtime_error("Can't read " + file_path);
+        }
+
+        if (tensor_descr.layout != "CN") {
+            binary_file.read(&tensor_data[b * input_size], input_size);
+        }
+        else {
+            for (int i = 0; i < tensor_descr.channels(); ++i) {
+                binary_file.read(&tensor_data[(i * batch_size + b) * sizeof(T)], sizeof(T));
+            }
+        }
+    }
+
+    return tensor;
+}
+
+Ort::Value get_tensor_from_image(const inputs::InputDescr &input_descr, int batch_size, int start_index) {
+    auto precision = utils::get_data_precision(input_descr.tensor_descr.type);
+    if (precision == utils::DataPrecision::FP32) {
```

Comment:
Yeah. it's a matter of taste whether to use `switch` statement or` if/else`, I think. Switch would be slightly shorter but I amn't sure about readability. And it would be tedious to replace to `switch` statements in here and other places,  so if you don't mind let's leave it as is.

Reply:
Up to you. If\else will grow faster with new options from my point of view.